### PR TITLE
Improve Kernel.Def

### PR DESCRIPTION
**Pull request by eksperimental**
added formatting styles, and replaced a `case` when a `cond` (according to my opinion) made more sense.

### CODE REVIEW

File: lib/elixir/lib/kernel/def.ex [hunk 1]
```diff
@@ -2,16 +2,16 @@ defmodule Kernel.Def do
...
+  Callback invoked at compile time for `defdelegate`.
   """
   def delegate(fun, opts) do
     append_first = Keyword.get(opts, :append_first, false)
 
     {name, args} =
-          case Macro.decompose_call(fun) do
-            {_, _} = pair -> pair
-            _ -> raise ArgumentError, "invalid syntax in defdelegate #{Macro.to_string(fun)}"
-          end
```
Comment (eksperimental) [inline lib/elixir/lib/kernel/def.ex]: let me know if you want me to undo this indentation change.

File: lib/elixir/lib/kernel/def.ex [hunk 2]
```diff
@@ -24,13 +24,14 @@ defmodule Kernel.Def do
...
   @doc """
-  Callback invoked at compile time for defstruct.
+  Callback invoked at compile time for `defstruct`.
   """
   def struct(module, fields) do
-    case fields do
-      fs when is_list(fs) -> :ok
-      other ->
-        raise ArgumentError, "struct fields definition must be list, got: #{inspect other}"
+    cond do
```
Comment (josevalim) [inline lib/elixir/lib/kernel/def.ex]: It needs to be a `case` because of bootstrap issues. :)

File: lib/elixir/lib/kernel/def.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/kernel/def.ex]: No, thanks for fixing it!

**Comment by eksperimental** [comment]
@josevalim we are good to go

**Pull request merged**